src/EventSubscriber/OgEventSubscriber.php

```diff
@@ -88,12 +167,125 @@ public function provideDefaultOgPermissions(PermissionEventInterface $event) {
    */
   public function provideDefaultRoles(DefaultRoleEventInterface $event) {
     /** @var \Drupal\og\Entity\OgRole $role */
-    $role = $this->ogRoleStorage->create([
+    $role = $this->entityTypeManager->getStorage('og_role')->create([
       'name' => OgRoleInterface::ADMINISTRATOR,
       'label' => 'Administrator',
       'is_admin' => TRUE,
     ]);
     $event->addRole($role);
   }
 
+  /**
+   * Returns a list of generic entity operation permissions for group content.
+   *
+   * This returns generic group content entity operation permissions for the
+   * operations 'create', 'update' and 'delete'.
+   *
+   * In Drupal the entity operation permissions are not following a machine
+   * writable naming scheme, but instead they use an arbitrary human readable
+   * format. For example the permission to update nodes of type article is 'edit
+   * own article content'. This does not even contain the operation 'update' or
+   * the entity type 'node'.
+   *
+   * OG needs to be able to provide basic CRUD permissions for its group content
+   * even if it cannot generate the proper human readable versions. This method
+   * settles for a generic permission format '{operation} {ownership} {bundle}
+   * {entity type}'. For example for editing articles this would become 'update
+   * own article node'.
+   *
+   * Modules can implement their own PermissionEvent to declare their proper
+   * permissions to use instead of the generic ones. For an example
+   * implementation, see `provideDefaultNodePermissions()`.
+   *
+   * @param array $group_content_bundle_ids
+   *   An array of group content bundle IDs, keyed by group content entity type
+   *   ID.
+   *
+   * @return \Drupal\og\GroupContentOperationPermission[]
+   *   The array of permissions.
+   *
+   * @see \Drupal\og\EventSubscriber\OgEventSubscriber::provideDefaultNodePermissions()
+   */
+  protected function getDefaultEntityOperationPermissions(array $group_content_bundle_ids) {
+    $permissions = [];
+
+    foreach ($group_content_bundle_ids as $group_content_entity_type_id => $bundle_ids) {
+      foreach ($bundle_ids as $bundle_id) {
+        $permissions += $this->generateEntityOperationPermissionList($group_content_entity_type_id, $bundle_id);
```

Comment:
Yes, first the permissions for all supported group content bundles are generated, and then they are overridden by the module-specific ones, like the Node module. This is setting the values twice for the Node module, but this doesn't really matter since it is a very cheap operation.